lib/service_actor/arguments_validator.rb

```diff
@@ -4,8 +4,11 @@ module ServiceActor::ArgumentsValidator
   module_function
 
   def validate_origin_name(name, origin:)
-    return if name.to_sym == :error
-    return unless ServiceActor::Result.instance_methods.include?(name.to_sym)
+    name = name.to_sym
+    return if name == :error
+
+    methods = ServiceActor::Core.instance_methods + ServiceActor::Result.instance_methods
```

Comment:
I'd probably add a test to ensure `ServiceActor::Core.instance_methods` stays the same (just like we do [here](https://github.com/sunny/actor/blob/68c2be1959bfeea6d07a578935b375c9cb5171b1/spec/service_actor/result_spec.rb#L19)) with a comment referencing this PR. 

So if we add a new instance method in future, we'll know we add another disallowed origin name

Reply:
Good call, I’ve added a spec for just that 👍🏻